crates/recursor/src/recursor.rs

```diff
@@ -568,6 +556,25 @@ impl Recursor {
     }
 }
 
+// as per section 3.2.1 of RFC4035
+fn maybe_strip_dnssec_records(query_has_dnssec_ok: bool, lookup: Lookup, query: Query) -> Lookup {
+    if !query_has_dnssec_ok {
```

Comment:
Let's invert the condition and do an early return here?

Reply:
done